pipeline/state_duration.go

```diff
@@ -0,0 +1,58 @@
+package pipeline
+
+import (
+	"time"
+
+	"github.com/influxdata/kapacitor/tick/ast"
+)
+
+// Compute the duration of a given state.
+// The state is defined via a lambda expression. For each consecutive point for
+// which the expression evaluates as true, the state duration will be
+// incremented by the duration between points. When a point evaluates as false,
+// the state duration is reset to 0.
+//
+// The state duration will be added as an additional field to each point. If the
+// expression evaluates as false, the field will not be added. If the expression
```

Comment:
We could make it default to `-1`, as this will satisfy the 99.9999% of use cases, and then just provide a `.default()` attribute to change it to whatever the user desires.